cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+        target_provider=None,
+        osp_security_group=None,
+        osp_flavor=None,
+        pre_playbook=None,
+        post_playbook=None,
+        start_migration=False,
+    ):
+        """Create new migration plan in UI
+        Args:
+            name: (string) plan name
+            description: (string) plan description
+            infra_map: (object) infra map object name
+            vm_list: (list) list of vm objects
+            csv_import: (bool) flag for importing vms
+            target_provider:Target provider (OSP or RHV)
+            osp_security_group: security group for OSP
+            osp_flavor:Flavor for OSP,
+            pre_playbook: (string) pre-migration playbook name
+            post_playbook: (string) post-migration playbook name
+            start_migration: (bool) flag for start migration
+        """
+        view = navigate_to(self, "Add")
+
+        radio_btn = None
+        if csv_import:
+            radio_btn = "Import a CSV file with a list of VMs to be migrated"
+        view.general.fill({"infra_map": infra_map,
+                           "name": name,
+                           "description": description,
+                           "csv_import": radio_btn})
+
+        view.vms.wait_displayed()
+        view.vms.fill({
+            'csv_import': csv_import,
+            'vm_list': vm_list})
+
+        # For OSP we need to fill this extra tab
+        if target_provider:
+            if target_provider.one_of(OpenStackProvider):
+                view.instance_properties.wait_displayed()
+                view.instance_properties.fill({
+                    'osp_security_group': osp_security_group,
+                    'osp_flavor': osp_flavor
+                })
+
+        view.advanced.fill({
+            'pre_playbook': pre_playbook,
+            'post_playbook': post_playbook
+        })
+
+        view.schedule.fill(start_migration)
+        return self.instantiate(name=name, infra_map=infra_map, vm_list=vm_list)
+
+    def is_plan_started(self, name):
+        view = navigate_to(self, "All")
+        return wait_for(
+            func=view.progress_card.is_plan_started,
+            func_args=[name],
+            message="migration plan is starting, be patient please",
+            delay=5,
+            num_sec=150,
+            handle_exception=True
+        )
+
+
+@navigator.register(MigrationPlanCollection, "All")
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute("appliance.server", "LoggedIn")
+    VIEW = MigrationPlanView
+
+    def step(self):
+        self.prerequisite_view.navigation.select("Compute", "Migration", "Migration Plans")
+
+
+@navigator.register(MigrationPlanCollection, "Add")
+class AddMigrationPlan(CFMENavigateStep):
+    VIEW = AddMigrationPlanView
+    prerequisite = NavigateToSibling("All")
+
+    def step(self):
+        self.prerequisite_view.create_migration_plan.click()
+
+
+@navigator.register(MigrationPlan, "NotStarted")
```

Comment:
These seem like collection navigation destinations, since they're not navigating to an entity, or a page of the entity specifically.  There's no reference to the contextual object, only to prerequisite view widgets. Consider registering them against both the entity and collection, if not the collection only.